src/dll/edbgEcmdDll.C

```diff
@@ -74,6 +74,9 @@ std::map<ecmdChipTarget, std::string, ecmdChipTargetCompare> eeproms;
 // Store if we have init'd the VPD info
 bool vpdInit = false;
 
+//chip type
+std::string chipType;
```

Comment:
Why a global variable with the chip type?

Reply:
I can do that. since init is called once in the program I thought it is the best place I can call and assign to the global variable. I just tried to optimize the calling to the getChipType() every time and instead use the global var instead. But, i can change it if you think it is wrong approach.

Reply:
I would rather we get the information from pdbg when needed if it has an API for it.  At a minimum, it should be in a returned by a function so it is easy to change in the future if need.  Also,  it should be an enum value and not string we have to compare all the time

Reply:
Done. I am using pdbg api only if you had checked the code. just that we need to get api to upstream and same will be done